docs/src/pages/ui/theme/alternativeStyling/index.page.mdx

```diff
@@ -0,0 +1,501 @@
+
+##### CSS Root Variable Overrides
+
+Amplify theming provides many CSS variables to customize the look and feel of the entire application. These CSS variables can be used at a root level to override all instances of component styling.
+
+```jsx
+import React from 'react';
+import styled from 'styled-components';
+import { Button } from '@aws-amplify/ui-react';
+
+const StyledButton = styled(Button)`
+  color: blue;
+`;
+
+<StyledButton>Button</StyledButton>;
+```
+
+```CSS
+/* External Style Sheet */
+:root {
+  --amplify-components-button-color: red;
+}
+```
+
+In the example above, the button color will be set to `blue` because the Styled Component will take precedence over the Amplify CSS variable.
+
+</Block>
+
+<Block name="JSS">
+[JSS](https://github.com/cssinjs/jss/tree/master/packages/react-jss) is a library for generating Style Sheets with Javascript.
+
+#### Usage
+
+```jsx
+import React from 'react';
+import { createUseStyles } from 'react-jss';
+import { View } from '@aws-amplify/ui-react';
+
+const useStyles = createUseStyles({
+  styledText: {
+    color: 'blue',
+  },
+});
+
+const classes = useStyles();
+
+//use the JSS create classes in the Amplify ui className
+<Text className={classes.styledText}>This is my JSS Text </Text>;
+```
+
+#### Interactions
+
+There are various ways to customize an Amplify component and JSS will interact with these customizations in the following ways:
+
+##### Amplify Styling Props
+
+These are styling props that can be used directly on an Amplify component which will affect a single style property. Examples of styling props are `color` or `fontWeight`. These Amplify styling props will take precedence over JSS styling.
+
+```jsx
+import React from 'react';
+import { createUseStyles } from 'react-jss';
+import { View } from '@aws-amplify/ui-react';
+
+const useStyles = createUseStyles({
+  styledText: {
+    color: 'blue',
+  },
+});
+
+const classes = useStyles();
+
+<Text className={classes.styledText} color="red">
+  This is my JSS Text
+</Text>;
+```
+
+In the example above, the color of the view will be set to `red` because of the Amplify styling prop. [Here]() is a list of all Amplify styling props.
+
+##### Amplify variation props
+
+These props change the look and/or behavior of certain Amplify components. Examples include `size` and `variation`.
+
+```jsx
+import React from 'react';
+import { createUseStyles } from 'react-jss';
+import { Button } from '@aws-amplify/ui-react';
+
+const useStyles = createUseStyles({
+  styledButton: {
+    color: 'blue',
+  },
+});
+
+const classes = useStyles();
+
+<Button className={classes.styledButton} variation="primary">
+  Primary Button
+</Button>;
+```
+
+In the example above, the color of the button will be set to `white`, because the primary variation of the button has a color set to `white` and it takes precedence over the JSS styling. A list of Amplify variations can be found on [here]().
```

Comment:
Does JSS support `&[data-attribute]` selectors?

Reply:
yes `&[data-attribute]` selectors are supported. In this guide I was aiming to just describe the default behavior and the experience I would expect developers would run into if they just use CSS in JS libraries without specifically trying to interact with amplify's styling